cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go

```diff
@@ -225,18 +266,23 @@ type BootstrapToken struct {
 	Token *BootstrapTokenString `json:"token"`
 	// Description sets a human-friendly message why this token exists and what it's used
 	// for, so other administrators can know its purpose.
+	// +optional
 	Description string `json:"description,omitempty"`
 	// TTL defines the time to live for this token. Defaults to 24h.
 	// Expires and TTL are mutually exclusive.
+	// +optional
 	TTL *metav1.Duration `json:"ttl,omitempty"`
```

Comment:
Is 0 a valida TTL in this case? Is there a minimum?

Reply:
this TTL is used to create the unstuctured format for the bootstrap token v1.Secret type:
https://kubernetes.io/docs/reference/access-authn-authz/bootstrap-tokens/#bootstrap-token-secret-format

so if the `expiration` is 0s from now, i'd assume that it would pass validation, but the token cleaner controller from core would just GC it on it's next cycle.